lib/bap_disasm/bap_disasm_x86_lifter.ml

```diff
@@ -0,0 +1,2319 @@
+     :: pf_e                       (*  2 *)
+     :: undefined 1                (*  1 *)
+     :: cf_e                       (*  0 *)
+     :: []
+let bap_to_eflags = List.drop bap_to_rflags 32
+let bap_to_flags = List.drop bap_to_eflags 16
+let bap_to_lflags = List.drop bap_to_flags 16
+
+let rflags_e = List.reduce_exn ~f:Exp.(^) bap_to_rflags
+let eflags_e = List.reduce_exn ~f:Exp.(^) bap_to_eflags
+let flags_e = List.reduce_exn ~f:Exp.(^) bap_to_flags
+let lflags_e = List.reduce_exn ~f:Exp.(^) bap_to_lflags
+
+let rflags_to_bap =
+  let assn v = Some (v, Util.id) in
+  (List.map ~f:(fun _ -> None) (List.range ~stride:(-1) ~stop:`inclusive 63 32))
+  @  None                       (* 31 *)
+     :: None                       (* 30 *)
+     :: None                       (* 29 *)
+     :: None                       (* 28 *)
+     :: None                       (* 27 *)
+     :: None                       (* 26 *)
+     :: None                       (* 25 *)
+     :: None                       (* 24 *)
+     :: None                       (* 23 *)
+     :: None                       (* 22 *)
+     :: None                       (* 21 *)
+     :: None                       (* 20 *)
+     :: None                       (* 19 *)
+     :: None                       (* 18 *)
+     :: None                       (* 17 *)
+     :: None                       (* 16 *)
+     :: None                       (* 15 *)
+     :: None                       (* 14 *)
+     :: None                       (* 13 *)
+     :: None                       (* 12 *)
+     :: assn oF                    (* 11 *)
+     :: assn df                    (* 10 *)
+     :: None                       (* 09 *)
+     :: None                       (* 08 *)
+     :: assn sf                    (* 07 *)
+     :: assn zf                    (* 06 *)
+     :: None                       (* 05 *)
+     :: assn af                    (* 04 *)
+     :: None                       (* 03 *)
+     :: assn pf                    (* 02 *)
+     :: None                       (* 01 *)
+     :: assn cf                    (* 00 *)
+     :: []
+let eflags_to_bap = List.drop rflags_to_bap 32
+let flags_to_bap = List.drop eflags_to_bap 16
+let lflags_to_bap = List.drop flags_to_bap 8
+(* A list of functions for assigning each bit in rflags *)
+let assns_rflags_to_bap =
+  List.map
+    ~f:(function
+        | None -> (fun _ -> [])
+        | Some (v,f) -> (fun e -> [Stmt.move v (f e)]))
+    rflags_to_bap
+let assns_eflags_to_bap = List.drop assns_rflags_to_bap 32
+let assns_flags_to_bap = List.drop assns_eflags_to_bap 16
+let assns_lflags_to_bap = List.drop assns_flags_to_bap 8
+
+(* exp helpers *)
+
+let load_s mode s t a =
+  let mem_e = ge mode mem in
+  match s with
+  | None -> Exp.load mem_e a LittleEndian t
+  | Some v -> Exp.(load mem_e (var v + a) LittleEndian t)
+
+(* exp from int64 *)
+(* we are not sure of the big int to bitvector stuff *)
+let bitvector_of_bil = function
```

Comment:
This function should be removed. All it occurrences are applied to a constant case, like:
`bitvector_of_bil (l32 i)`, that can be refactored to `BV.of_int64 ~width:32 i`
You should consider adding helpers, that will do the proper casts without further burden, like `bv32_t : int64 -> word`, and `bv64_t : int64 -> word`.